src/main/java/teammates/ui/controller/AdminEmailImageUploadAction.java

```diff
@@ -1,103 +1,50 @@
 package teammates.ui.controller;
 
-import java.util.List;
-import java.util.Map;
-
+import teammates.common.util.AppUrl;
 import teammates.common.util.Config;
 import teammates.common.util.Const;
 import teammates.logic.api.GateKeeper;
 
 import com.google.appengine.api.blobstore.BlobInfo;
 import com.google.appengine.api.blobstore.BlobKey;
-import com.google.appengine.api.blobstore.BlobstoreFailureException;
-import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
 
-public class AdminEmailImageUploadAction extends Action {
+public class AdminEmailImageUploadAction extends ImageUploadAction {
     
     AdminEmailComposePageData data;
     
     @Override
     protected ActionResult execute() {
```

Comment:
Previously I asked whether this method was a duplicate that should be removed and your response was that this overrides its parent method. Still, even if not removed, both methods are more than 50% similar (as a proof, when I asked to refactor to use `AppUrl`, the change has to be repeated twice).

Is it absolutely necessary for those similarity to be duplicated in both `ImageUploadAction` and its admin email counterpart? Ideally (albeit may not be possible) only `ImageUploadAction` contains the method definition of `execute`, with appropriate parts to be overridden by its admin email counterpart.